src/Val.cc

```diff
@@ -1889,45 +1889,54 @@ ValPtr TableVal::Default(const ValPtr& index) {
         return nullptr;
     }
 
+    ValPtr result;
+
     if ( def_val->GetType()->Tag() != TYPE_FUNC || same_type(def_val->GetType(), GetType()->Yield()) ) {
         if ( def_attr->GetExpr()->IsConst() )
             return def_val;
 
         try {
-            return def_val->Clone();
-        } catch ( InterpreterException& e ) { /* Already reported. */
+            result = def_val->Clone();
+        } catch ( InterpreterException& e ) {
+            Error("&default value for table is not clone-able");
```

Comment:
I think my suggestion failed :-) I'll move this one block out and keep the `if ( ! result )` to handle the situation where `->Clone()` returns nil. If I read things right, would result in a segfault on the `result->GetType()` further down now.